docs/debug.md

```diff
@@ -0,0 +1,137 @@
+# Debugging tools
+
+Playwright scripts work with existing debugging tools, like Node.js debuggers
+and browser developer tools. Playwright also introduces new debugging features
+for browser automation.
+
+<!-- GEN:toc -->
+- [Run in headful mode](#run-in-headful-mode)
+- [Visual Studio Code debugger](#visual-studio-code-debugger)
+  * [Use launch config](#use-launch-config)
+  * [Use the new JavaScript debugger](#use-the-new-javascript-debugger)
+- [Browser Developer Tools](#browser-developer-tools)
+  * [API for Chromium](#api-for-chromium)
+- [Run in Debug Mode](#run-in-debug-mode)
+  * [Defaults](#defaults)
+  * [Debugging Selectors](#debugging-selectors)
+  * [Evaluate Source Maps](#evaluate-source-maps)
+- [Verbose API logs](#verbose-api-logs)
+<!-- GEN:stop -->
+
+## Run in headful mode
+
+Playwright runs browsers in headless mode by default. To change this behavior,
+use `headless: false` as a launch option. You can also use the `slowMo` option
+to slow down execution and follow along while debugging.
+
+```js
+await chromium.launch({ headless: false, slowMo: 100 }); // or firefox, webkit
+```
+
+## Visual Studio Code debugger
+
+The VS Code debugger can be used to pause and resume execution of Playwright
+scripts with breakpoints. The debugger can be configured in two ways.
+
+### Use launch config
+
+Setup [`launch.json` configuration](https://code.visualstudio.com/docs/nodejs/nodejs-debugging)
+for your Node.js project. Once configured launch the scripts with F5 and use
+breakpoints.
+
+### Use the new JavaScript debugger
+
+VS Code 1.46+ introduces the new JavaScript debugger behind a feature flag. The
+new debugger does not require a `launch.json` configuration. To use this:
+
+1. Enable the preview debugger
+    * Open JSON settings and add `"debug.javascript.usePreview": true`
+    * Open settings UI and enable the `Debug › JavaScript: Use Preview` setting
+
+1. Set a breakpoint in VS Code
+    * Use the `debugger` keyword or set a breakpoint in the VS Code UI
+
+1. Run your Node.js script from the terminal
```

Comment:
Shouldn't this be a special debugging terminal?

Reply:
from what I can tell, vscode 1.46+ does not require a special terminal